sphinx/ext/javadoctest.py

```diff
@@ -0,0 +1,118 @@
+"""Mimic javadoctest in Sphinx.
+
+The extension automatically execute code snippets and checks their results on Java engine.
+"""
+
+import subprocess
+from typing import Any, Dict
+
+from sphinx.application import Sphinx
+from sphinx.ext.doctest import (
+    DocTestBuilder,
+    TestcodeDirective,
+    TestoutputDirective,
+    doctest,
+    sphinx,
+)
+from sphinx.locale import __
+
+
+class JavaTestcodeDirective(TestcodeDirective):
+    def run(self):
+        node_list = super().run()
+        print("check_node_list")
+        print(node_list)
+        node_list[0]["language"] = "java"
+        return node_list
+
+
+class JavaDocTestBuilder(DocTestBuilder):
+    """
+    Runs Java test snippets in the documentation.
+    """
+
+    name = "javadoctest"
+    epilog = __(
+        "Java testing of doctests in the sources finished, look at the "
+        "results in %(outdir)s/output.txt.",
+    )
+
+    def compile(
+        self, code: str, name: str, type: str, flags: Any, dont_inherit: bool,
+    ) -> Any:
+        if self.config.javadoc_options['flavor'].lower() == 'java':
+            stdout_dependency = ''
+        elif self.config.javadoc_options['flavor'].lower() == 'java_with_maven':
+            # go to project that contains all your Java maven dependencies
+            path_java_project = self.config.javadoc_options['path']
+            # create list of all Java jar dependencies
+            subprocess.check_call(
+                [
+                    "mvn",
+                    "-q",
+                    "dependency:build-classpath",
+                    "-DincludeTypes=jar",
+                    "-Dmdep.outputFile=.cp.tmp",
+                    f"-Dversion={self.env.config.version}",
+                ],
+                cwd=path_java_project,
+                text=True,
+            )
+            if not (path_java_project / ".cp.tmp").exists():
+                raise RuntimeError(
```

Comment:
Maybe another type of exception (e.g., derived from ExtensionError?)

Reply:
Changed